File: frontend/src/types/productTYpes/productResponse.ts
export interface ProductResponse {
  id?: number;
  sku: string;
  name: string;
  description?: string | null;
  price: number;
  quantity: number;
  weightKg?: number | null;
  imageUrl?: string | null;
  categoryId?: number | null;
  warehouseId?: number | null;
  active: boolean;
}


export interface CreateProductRequest {
  sku: string;
  name: string;
  description?: string;
  price: number;
  quantity: number;
  weightKg?: number | null;
  imageUrl?: string | null;
  categoryId?: number | null;
  warehouseId?: number | null;
}

File: product-service/src/main/java/com/logistics/product_service/dto/response/ProductResponse.java
package com.logistics.product_service.dto.response;

import lombok.Data;

import java.math.BigDecimal;

@Data
public class ProductResponse {

    private Long id;

    private String sku;

    private String name;

    private String description;

    private BigDecimal price;
    private int quantity;

    private Double weightKg;

    private String imageUrl;

    private Long categoryId;

    private Long warehouseId;

    private boolean active;
}
